### PR TITLE
Add Implementation for CF-912E

### PR DESCRIPTION
**Pull request by DenialRiver1434**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/gold/cf-912E.mdx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+vector<ll> onums, enums;
+
+bool test(ll x) {
+	ll tot = 0;
+
+	// Two pointers
+	ll opoint = lower_bound(onums.begin(), onums.end(), x) - onums.begin();
+	if (opoint == onums.size()) opoint--;
+
+	for (auto e : enums) {
```
Comment (SansPapyrus683) [inline solutions/gold/cf-912E.mdx:36]: can you not use auto

File: solutions/gold/cf-912E.mdx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+			hi = mid;
+		} else {
+			lo = mid + 1;
+		}
+	}
+	return lo;
+}
+
+// Run DFS to generate all possible products
+
```
Comment (SansPapyrus683) [inline solutions/gold/cf-912E.mdx:60]:
```suggestion
```

File: solutions/gold/cf-912E.mdx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+void dfsa(ll x, ll pos) {
+	onums.push_back(x);
+	for (int i = pos; i < odds.size(); i++) {
+		if (MAX / x < odds[i]) break;  // prevent overflow
```
Comment (SansPapyrus683) [inline solutions/gold/cf-912E.mdx:64]: braces

File: solutions/gold/cf-912E.mdx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+		dfsa(x * odds[i], i);
+	}
+}
+
+void dfsb(ll x, ll pos) {
```
Comment (SansPapyrus683) [inline solutions/gold/cf-912E.mdx:69]: plz use better names other than `dfsa` and `dfsb`

File: solutions/gold/cf-912E.mdx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+int main() {
+	cin >> N;
+	for (int i = 0; i < N; i++) {
+		ll x;
+		cin >> x;
+		if (i % 2 == 0) odds.push_back(x);
+		else evens.push_back(x);
+	}
+	cin >> K;
```
Comment (SansPapyrus683) [inline solutions/gold/cf-912E.mdx:85]: rename to `n` and `k`

File: solutions/gold/cf-912E.mdx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+	// Pre-generate everything
+	dfsa(1, 0);
+	dfsb(1, 0);
+	sort(onums.begin(), onums.end());
+	sort(enums.begin(), enums.end());
+
+	test(9);
```
Comment (SansPapyrus683) [inline solutions/gold/cf-912E.mdx:93]: magic #

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/gold/cf-912E.mdx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+	}
+}
+
+int main() {
+	cin >> n;
+	for (int i = 0; i < n; i++) {
+		ll x;
+		cin >> x;
+		if (i % 2 == 0) odds.push_back(x);
+		else evens.push_back(x);
```
Comment (SansPapyrus683) [inline solutions/gold/cf-912E.mdx:82]: braces

File: solutions/gold/cf-912E.mdx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+	}
+	cin >> k;
```
Comment (SansPapyrus683) [inline solutions/gold/cf-912E.mdx:84]:
```suggestion

	cin >> k;
```

File: solutions/gold/cf-912E.mdx [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+		if (e > x) break;
+		while (onums[opoint] > x / e) {  // Written to prevent overflow
+			opoint--;
+		}
+		tot += opoint + 1;
+	}
+	return tot >= k;
+}
+
+ll first_true(ll lo, ll hi) {
```
Comment (SansPapyrus683) [inline solutions/gold/cf-912E.mdx:46]: add link to binsearch module in a doc comment

File: solutions/gold/cf-912E.mdx [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+**Time Complexity:** $\mathcal{O}(\log 10^{18} \cdot (|D(A)| + |D(B)|))$
+
+<LanguageSection>
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+using ll = long long;
+const ll MAX = 1e18;
```
Comment (SansPapyrus683) [inline solutions/gold/cf-912E.mdx:21]: max what?

File: solutions/gold/cf-912E.mdx [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+[Official Editorial](https://codeforces.com/blog/entry/56920)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(\log 10^{18} \cdot (|D(A)| + |D(B)|))$
+
+<LanguageSection>
+<CPPSection>
+```cpp
```
Comment (SansPapyrus683) [inline solutions/gold/cf-912E.mdx:16]:
```suggestion

```cpp
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/gold/cf-912E.mdx [hunk 12]
```diff
@@ -0,0 +1,96 @@
+---
+id: cf-912E
+source: CodeForces
+title: Prime Gift
+author: Daniel Ge
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/56920)
```
Comment (ryanchou-dev) [inline solutions/gold/cf-912E.mdx:8]:
```suggestion
[Official Editorial (C++)](https://codeforces.com/blog/entry/56920)
```

File: solutions/gold/cf-912E.mdx [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(\log 10^{18} \cdot (|D(A)| + |D(B)|))$
```
Comment (ryanchou-dev) [inline solutions/gold/cf-912E.mdx:12]: constant factors are omitted from big O right?

maybe add an `<Asterisk>` about it or a small note if it contributes a huge chunk of it?

File: solutions/gold/cf-912E.mdx [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+	cin >> n;
+	for (int i = 0; i < n; i++) {
+		ll x;
+		cin >> x;
+		if (i % 2 == 0) odds.push_back(x);
+		else evens.push_back(x);
+	}
+	cin >> k;
+
+	// Pre-generate everything
```
Comment (ryanchou-dev) [inline solutions/gold/cf-912E.mdx:86]: elaborate?

File: solutions/gold/cf-912E.mdx (hunk 3 above)
Comment (ryanchou-dev) [reply solutions/gold/cf-912E.mdx:63]: not fixed?

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]

File: solutions/gold/cf-912E.mdx (hunk 3 above)
Comment (DenialRiver1434) [reply solutions/gold/cf-912E.mdx:64]: sorry I only fixed second one

File: solutions/gold/cf-912E.mdx [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+		if (e > x) break;
+		while (onums[opoint] > x / e) {  // Written to prevent overflow
+			opoint--;
+		}
+		tot += opoint + 1;
+	}
+	return tot >= k;
+}
+
+/*
```
Comment (SansPapyrus683) [inline solutions/gold/cf-912E.mdx:49]:
```suggestion
/**
```

**Review by SansPapyrus683 — APPROVED** [review]

**Review by ryanchou-dev — APPROVED** [review]

**Comment by ryanchou-dev** [comment]
exquisite

**Pull request merged**